libr/include/r_types.h

```diff
@@ -654,4 +654,16 @@ static inline void r_run_call10(void *fcn, void *arg1, void *arg2, void *arg3, v
 # endif
 #endif
 
+// reference counter
+typedef void (*RFreeCallback)(void *a);
+typedef struct r_ref_t {
+	volatile size_t count;
```

Comment:
why `volatile`?

Reply:
because it may be accessed from different threads